x-pack/plugins/infra/public/containers/metadata/with_metadata.tsx

```diff
@@ -58,7 +60,7 @@ export const WithMetadata = ({
 
 const getFilteredLayouts = (
   layouts: InfraMetricLayout[],
-  metadata: Array<MetadataQuery.MetadataByNode | null> | undefined
+  metadata: Array<MetadataQuery.Features | null> | undefined
```

Comment:
If we disallowed `null` values in the schema as proposed below, we wouldn't need to include it here. The `undefined` is also handled by line 49 above.

```suggestion
  metadata: MetadataQuery.Features[]
```

Reply:
How about also representing the argument type change in the name, e.g. `nodeFeatures` instead of `metadata`?